source/client-side-operations-timeout/tests/global-timeoutMS.yml

```diff
@@ -0,0 +1,3142 @@
+          isTimeoutError: true
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: aggregate
+              databaseName: *databaseName
+              command:
+                aggregate: *collectionName
+                maxTimeMS: { $$type: ["int", "long"] }
+  - description: "timeoutMS can be set to 0 on a MongoClient - aggregate on collection"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                uriOptions:
+                  timeoutMS: 0
+                useMultipleMongoses: false
+                observeEvents:
+                  - commandStartedEvent
+                ignoreCommandMonitoringEvents:
+                  - killCursors
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: ["aggregate"]
+              blockConnection: true
+              blockTimeMS: 15
+      - name: aggregate
+        object: *collection
+        arguments:
+          pipeline: []
+          
+        
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: aggregate
+              databaseName: *databaseName
+              command:
+                aggregate: *collectionName
+                maxTimeMS: { $$exists: false }
+  - description: "timeoutMS can be configured on a MongoClient - count on collection"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                uriOptions:
+                  timeoutMS: 50
+                useMultipleMongoses: false
+                observeEvents:
+                  - commandStartedEvent
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: ["count"]
+              blockConnection: true
+              blockTimeMS: 60
+      - name: count
```

Comment:
There is no `count` command in the Go driver (nor in a couple other, new drivers), so I think we'll skip tests with `count`.

Reply:
Yep drivers without the deprecated count helper will skip tests for count. Note there are already tests for count:
```
git grep ' name: count$'
source/client-side-encryption/etc/test-templates/count.yml.template:19:      - name: count
source/client-side-encryption/etc/test-templates/count.yml.template:50:      - name: count
source/client-side-encryption/tests/count.yml:19:      - name: count
source/client-side-encryption/tests/count.yml:50:      - name: count
source/crud/tests/v1/read/count-collation.yml:20:            name: count
source/crud/tests/v1/read/count-empty.yml:24:            name: count
source/crud/tests/v1/read/count.yml:48:            name: count
source/crud/tests/v1/read/count.yml:57:            name: count
source/crud/tests/v1/read/count.yml:67:            name: count
source/retryable-reads/tests/legacy/count-serverErrors.yml:26:                    name: count
source/retryable-reads/tests/legacy/count.yml:22:                    name: count
source/transactions/tests/legacy/count.yml:25:      - name: count
```

Reply:
cc: @kmahar re: https://github.com/mongodb/specifications/pull/1206#discussion_r865129646

Reply:
As described in that linked thread, I'll add the logic in Go to automatically skip tests with unsupported operations. Other drivers that decode entire test files and the contained operations into native types will likely find issue with the `count` and `listIndexNames` tests.